File: xamarin/Xtrade/Xtrade.AndroidApp/Adapters/ViewHolders/RateViewHolder.cs
namespace Xtrade.AndroidApp.Adapters.ViewHolders
{
    using System;
    using Android.Support.V7.Widget;
    using Android.Views;
    using Android.Widget;

    public class RateViewHolder : RecyclerView.ViewHolder
    {
        public ImageView ForexFlagImageView { get; set; }

        public TextView ForexCodeTextView { get; set; }

        public TextView BuyRateTextView { get; set; }

        public TextView SellRateTextView { get; set; }

        public RateViewHolder(View itemView, Action<int> itemClickAction) : base(itemView)
        {
            this.ForexFlagImageView = itemView.FindViewById<ImageView>(Resource.Id.forexFlagImageView);
            this.ForexCodeTextView = itemView.FindViewById<TextView>(Resource.Id.forexCodeTextView);
            this.BuyRateTextView = itemView.FindViewById<TextView>(Resource.Id.buyRateTextView);
            this.SellRateTextView = itemView.FindViewById<TextView>(Resource.Id.sellRateTextView);
            this.ForexCodeTextView = itemView.FindViewById<TextView>(Resource.Id.forexCodeTextView);

            if (itemClickAction != null)
            {
                itemView.Click += (sender, args) =>
                {
                    itemClickAction(this.AdapterPosition);
                };
            }
        }
    }
}

File: android/Xtrade/app/src/main/java/nz/co/fendustries/xtrade/adapters/viewHolders/RateViewHolder.java
package nz.co.fendustries.xtrade.adapters.viewHolders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import nz.co.fendustries.xtrade.R;
import nz.co.fendustries.xtrade.interfaces.RecyclerViewItemTapCallback;

/**
 * Created by joshuafenemore on 21/10/16.
 */
public class RateViewHolder extends RecyclerView.ViewHolder
{
    private ImageView forexFlagImageView;
    private TextView forexCodeTextView;
    private TextView buyRateTextView;
    private TextView sellRateTextView;

    public RateViewHolder(View itemView, final RecyclerViewItemTapCallback recyclerViewItemTapCallback)
    {
        super(itemView);
        this.forexFlagImageView = (ImageView) itemView.findViewById(R.id.forexFlagImageView);
        this.forexCodeTextView = (TextView) itemView.findViewById(R.id.forexCodeTextView);
        this.buyRateTextView = (TextView) itemView.findViewById(R.id.buyRateTextView);
        this.sellRateTextView = (TextView) itemView.findViewById(R.id.sellRateTextView);

        itemView.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                if (recyclerViewItemTapCallback != null)
                {
                    recyclerViewItemTapCallback.onItemTap(getAdapterPosition());
                }
            }
        });
    }

    public ImageView getForexFlagImageView()
    {
        return forexFlagImageView;
    }

    public TextView getForexCodeTextView()
    {
        return forexCodeTextView;
    }

    public TextView getBuyRateTextView()
    {
        return buyRateTextView;
    }

    public TextView getSellRateTextView()
    {
        return sellRateTextView;
    }
}